### ISSUE TITLE
Not all package information in the output

### ISSUE THREAD

**Issue by U52284** [sensslen/nuget-license#32]
Whe I'm analyzing our project im only getting a few things and some are missing.

received:
```json
{
    "PackageId": "System.Text.RegularExpressions",
    "PackageVersion": "4.3.1",
    "PackageProjectUrl": "https://dot.net/",
    "License": "MS-EULA",
    "LicenseInformationOrigin": 1
}
```

expected:
```json
{
    "PackageId": "System.Text.RegularExpressions",
    "PackageVersion": "4.3.1",
    "PackageProjectUrl": "https://dot.net/",
    "**RepositoryUrl**": "https://github.com/dotnet/runtime",
    "License": "MS-EULA",
    "**LicenseUrl**": 
    "**LicenseFile**":
    "LicenseInformationOrigin": 1
  }
```

LicesneUrl is legacy, maybe its worth merging them having LicenseFile higher prio.

I saw that there are properties in the metadata but it seems they are not used.

**Comment by sensslen** [comment]
The information you get is exactly the information that is expected. The url is used to map to a license using a map. The licenseinformationorigin field is there to allow you to determine where the information was obtained from.

Please describe your use case why you need the additional information....

**Comment by U52284** [comment]
The usecase I have is that I have license information from npm as well as from nuget. for dotnet I'm using the nuget-license, but to get the license text, I need the license url or alternativly the repository url where I can also scrap the license file and build the license url myself. currently a lot of the projects have their project url set but I have to manually do some of the mapping of PackageX => LicenseUrlX.

I'm not understanding what the LicenseInformationOrigin actually is. Is this if you get if from teh repo, the mapping file or so?
I can't use the existing mechanisms becasue i have another source of license information.

**Comment by sensslen** [comment]
Maybe https://github.com/sensslen/nuget-license/blob/7e0ce5a52a2944325106c91c5515aedaeb23789a/src/NuGetUtility/LicenseValidator/LicenseInformationOrigin.cs helps to understand the LicenseInformationOrigin field. 

If the license is specified via a URL you can ask the tool to download the license file for you. However for all other scenarios you'd have to rely on the License field to then reach out to https://spdx.dev/ in order to get the license.

Feel free to open a PR that uses https://spdx.dev/ to also download licenses specified by other origins than url.....

I'm not positive to add the fields you requested, as this tool is not ment to give access to the nuget information but to validate whether the licenses used in a project are within a set of allowed licenses.

**Comment by U52284** [comment]
Yeah it's done internally with the -d parameter. Just wanted to do the download it myself, not with the parameter since I'm downloading it inmemory, not to files and the assemblying it as one license file. And since the information is already exists...

If the LicenseUrl is exposed I actually dont need the repository, this was just a solution to find the license url myself.

I thought the license Url should be part of the output since it is also part of the domain of the license, isn't it?

**Comment by sensslen** [comment]
I would argue that the output should follow a narrow model without many optional fields. Further the url to license mapping as well as the download already exist. Honestly I don't see a real benefit in allowing the licenses to be processed in memory only. It's easy enough to concatenate the files with a script.

Downloading the license texts by the license expression sounds like a nice addition though. As I said: feel free to add something like this...

**Comment by sensslen** [comment]
Implementing spdx license download should be easy enough using https://learn.microsoft.com/en-us/nuget/nuget-org/licenses.nuget.org

**Comment by U52284** [comment]
> I would argue that the output should follow a narrow model without many optional fields. Further the url to license mapping as well as the download already exist. Honestly I don't see a real benefit in allowing the licenses to be processed in memory only. It's easy enough to concatenate the files with a script.
> 
> Downloading the license texts by the license expression sounds like a nice addition though. As I said: feel free to add something like this...

You told me my use case and i think its common that applications have multiple sources of licenses as soon as they have more then one technology involved. I'm just trying bring the two together. 

But it seems I have to go with my solution I got now. But I will check that spdx you directed me to. maybe this will allow me some shortcuts as well.

**Comment by U52284** [comment]
I'm aware of that. But before using doing all this I want to know if the PR was doomed even before doing it so I don't wast time. Yours and mine.

I will try to make a PR then.

**Comment by sensslen** [comment]
@U52284 can you please check whether https://www.nuget.org/packages/nuget-license/3.0.3-beta.4 resolves the issue?

**Comment by garcipat** [comment]
It seems to work as expected. Texted it with that beta version.
